os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -462,6 +470,34 @@ void up_assert(const uint8_t *filename, int lineno)
 #else
 	lldbg("Assertion failed at file:%s line: %d\n", filename, lineno);
 #endif
+
+#ifdef CONFIG_DEBUG_WORKQUEUE
+	struct wqueue_s *workq;
+	bool is_asserted_workq = false;
+#ifdef CONFIG_SCHED_HPWORK
+	if (strncmp(this_task()->name, HPWORKNAME, sizeof(HPWORKNAME)) == 0) {
+		workq = (struct wqueue_s *)get_hpwork();
+		is_asserted_workq = true;
+	}
+#endif
+#ifdef CONFIG_SCHED_LPWORK
+	if (strncmp(this_task()->name, LPWORKNAME, sizeof(HPWORKNAME)) == 0) {
+		workq = (struct wqueue_s *)get_hpwork();
+		is_asserted_workq = true;
+	}
+#endif
+
+#ifdef CONFIG_SCHED_LPWORK
+	if (strncmp(this_task()->name, "uwork", sizeof("uwork")) == 0) {
+		workq = (struct wqueue_s *)get_usrwork();
+		is_asserted_workq = true;
+	}
+#endif
+	if (is_asserted_workq == true) {
+		lldbg("Running work function is %x, from pid %d.\n", workq->q.head, workq->worker[0].pid);
```

Comment:
`work->q.head` is not the function. 
It is `(work->q.head)->worker`. could you do double check?